### ISSUE TITLE
AttributeError: 'GPT4All' object has no attribute 'model_type'

### ISSUE THREAD

**Issue by dotandpixel** [imartinez/privateGPT#622]
**Describe the bug and how to reproduce it**
Following the setup instructions `README.md`. First time running the step `python privateGPT.py` after a successful `ingest.py`, the error is as follows:

```
% python privateGPT.py
Found model file.
gptj_model_load: loading model from 'models/ggml-gpt4all-j-v1.3-groovy.bin' - please wait ...
gptj_model_load: n_vocab = 50400
gptj_model_load: n_ctx   = 2048
gptj_model_load: n_embd  = 4096
gptj_model_load: n_head  = 16
gptj_model_load: n_layer = 28
gptj_model_load: n_rot   = 64
gptj_model_load: f16     = 2
gptj_model_load: ggml ctx size = 5401.45 MB
gptj_model_load: kv self size  =  896.00 MB
gptj_model_load: ................................... done
gptj_model_load: model size =  3609.38 MB / num tensors = 285
Traceback (most recent call last):
  File "/Users/user/.pyenv/versions/3.11.3/envs/vikings/privateGPT/privateGPT.py", line 76, in <module>
    main()
  File "/Users/user/.pyenv/versions/3.11.3/envs/vikings/privateGPT/privateGPT.py", line 36, in main
    llm = GPT4All(model=model_path, n_ctx=model_n_ctx, backend='gptj', callbacks=callbacks, verbose=False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic/main.py", line 1102, in pydantic.main.validate_model
  File "/Users/user/.pyenv/versions/vikings/lib/python3.11/site-packages/langchain/llms/gpt4all.py", line 156, in validate_environment
    values["backend"] = values["client"].model_type
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'GPT4All' object has no attribute 'model_type'
```

**Expected behavior**
Running `python privateGPT.py` for the first time after successful installation, expecting to see the text `> Enter your query`

**Environment**
 - MacOS 13.4 M1
 - Python 3.11.3
 - pyenv virtual

**Additional context**
`.env`
```
PERSIST_DIRECTORY=db
MODEL_TYPE=GPT4All
MODEL_PATH=models/ggml-gpt4all-j-v1.3-groovy.bin
EMBEDDINGS_MODEL_NAME=all-MiniLM-L6-v2
MODEL_N_CTX=1000
TARGET_SOURCE_CHUNKS=4

```

**Comment by beriagraphics** [comment]
Already fixed here [#5657](https://github.com/hwchase17/langchain/pull/5657)

**Comment by dotandpixel** [comment]
Langchain Issue: [#5720](https://github.com/hwchase17/langchain/issues/5720)
